### ISSUE TITLE
{{auto}} IP address seems to be broken

### ISSUE THREAD

**Issue by drjayvee** [getsentry/sentry-php#1215]
### Environment

How do you use Sentry?
Sentry SaaS

Which SDK and version?
PHP `3.1.0`

### Steps to Reproduce
According to the ["Identify Users" docs](https://docs.sentry.io/platforms/php/enriching-events/identify-user/):
> Set to "{{auto}}" to let Sentry infer the IP address from the connection.

However, looking at `Sentry\UserDataBag`, there's no automatic translation going on. In fact, [`setIpAddress(?string $ipAddress)`](https://github.com/getsentry/sentry-php/blob/master/src/UserDataBag.php#L175), will throw an `InvalidArgumentException` if the argument isn't a valid IP address.

I haven't actually tested it, but this looks to be broken (or the docs are wrong).

**Comment by stayallive** [comment]
> Set to "{{auto}}" to let Sentry infer the IP address from the connection.

I _think_ but I could be wrong here that this means it will infer it from the actual connection to sentry.io from the server the error is reported on so there won't be any client side translation.

So this is also not a very useful option for PHP applications since you probably want the IP of your users connection and not the IP of the server running the PHP code.

I'm sure this is a new option and we would have to allow that specific value I guess.

**Comment by drjayvee** [comment]
Since this is all about "identifying users", this should absolutely be the *request's remote IP address*, not the server's.

This is easily obtained with `$_SERVER['REMOTE_ADDR']` in PHP.

**Comment by stayallive** [comment]
Yes, but that is _not_ achieved using `{{auto}}`, that what I meant.

So if you want to get the user IP this is already collected for you, and you do not have to set the IP to anything for it. Setting it to `{{auto}}` will instruct the Sentry server to fill it with the connection IP.

If you want to retrieve the connection IP we have a integration that is enabled by default that does this for your: https://github.com/getsentry/sentry-php/blob/1b4acb089c63246d1a6f074e4b5ebcfe80be2057/src/Integration/RequestIntegration.php#L134-L148.

Keep in mind the above integration only works if you set [`send_default_pii`](https://docs.sentry.io/platforms/php/configuration/options/#send-default-pii) to `true` in the `init` call.

**Comment by Jean85** [comment]
This seems more of a documentation bug to me, it doesn't make sense to me to implement this behaviour for this SDK...

**Comment by stayallive** [comment]
This is server side. Not SDK side (except for allowing that exact value in the IP field).

**Comment by rhcarvalho** [comment]
> This seems more of a documentation bug to me, it doesn't make sense to me to implement this behaviour for this SDK...

I'll confirm this on Monday. I also think it is a documentation bug, we use `{{auto}}` for JavaScript and some other platforms, might have gotten replicated to PHP accidentally.

And yes, the `{{auto}}` placeholder gets replaced by [Relay](https://github.com/getsentry/relay) when ingesting the event.
